src/builder/specification.md

```diff
@@ -0,0 +1,339 @@
+# Builder API
+
+This document specifies the Builder API methods that the Consensus Layer uses to interact with external block builders.
+
```

Comment:
it may be worth adding a note that "builders" in this context are *trusted*

so that we can drop a lot of validations you would otherwise want to do

e.g. we can assume the execution header the validator gets back in `builder_getHeaderV1` is filled out in a way that reflects the validator's wishes so do not need to specify that the header is validated before signing over it bc builders in this iteration are trusted to do their job well (with a corresponding hit to reputation when they fail)


if we move to more decentralized alternatives it will likely become more important to validate more of the data we receive

Reply:
I agree, but I think it is important to still have an adversarial mindset when interacting with builders. If there areas of significant trust, we should work to reduce it. IMO the validator should validate the header they receive to the best of their ability before signing.